sdk/logs/src/main/java/io/opentelemetry/sdk/logs/data/LogDataBuilder.java

```diff
@@ -27,11 +28,26 @@
   @Nullable private String severityText;
   @Nullable private String name;
   private Body body = Body.stringBody("");
+  private final Clock clock;
   private final AttributesBuilder attributeBuilder = Attributes.builder();
 
-  LogDataBuilder(Resource resource, InstrumentationLibraryInfo instrumentationLibraryInfo) {
+  private LogDataBuilder(
+      Resource resource, InstrumentationLibraryInfo instrumentationLibraryInfo, Clock clock) {
     this.resource = resource;
     this.instrumentationLibraryInfo = instrumentationLibraryInfo;
+    this.clock = clock;
+  }
+
+  /** Returns a new {@link LogDataBuilder} with the default clock. */
+  public static LogDataBuilder create(
```

Comment:
Not for this PR I guess but I think we want a `LogDataBuilder` interface so that we can keep the implementation (and notably this method) private. I believe it's part of #3807 though

Reply:
Cool, yeah, follow-up effort I think too.